lib/__tests__/newClient.test.js

```diff
@@ -720,6 +720,103 @@ describe("Client", () => {
     })
   })
 
+  it("noops add authorizer when the client authorizes them self", async() => {
+    const addAuthorizer = await client1.addAuthorizer(
+      client1.config.clientId,
+      'sharedType'
+    )
+    // if response status successful, returns Promise that resolves true
+    expect(addAuthorizer).toEqual(true)
+  })
+
+  it("add a key and a policy to add an authorizer", async() => {
+    fetch.mockResponses(
+      [
+        JSON.stringify({
+          access_token: "getEAK",
+          expires_at: "xxxxx",
+          authorizer_public_key: { curve25519: "xxx" }
+        })
+      ],
+      [
+        JSON.stringify({
+          clientId: "responseClientId",
+          public_key: { curve25519: "xxxx" },
+          validated: true
+        })
+      ],
+      [
+        JSON.stringify({
+          access_token: "putNewEAK",
+          expires_at: "xxxxx",
+          authorizer_public_key: { curve25519: "xxx" }
+        })
+      ],
+      [JSON.stringify({ body: "" })]
+    )
+
+
+    const addAuthorizer = await client1.addAuthorizer(
+      "clientIdToAuthorize",
+      "recordTypeToAuthorize"
+    )
+
+    // auth token already cached
+    // tries to GET access key, succeeds
+    expect(fetch.mock.calls[0][0]).toEqual(
```

Comment:
These tests are hard to reason about -- could you add a "key" re: what the different indices mean in a comment?